### DISCUSSION TITLE
Warning on implicit conversion of `Unit` type to `Object`

### DISCUSSION THREAD

**Discussion by jl0pd** [fsharp/fslang-suggestions#1237] [Type System]
F# allows to pass unit `()` to methods accepting `object`. It's not desired in most cases since it 
1. Looks like regular method invocation
2. `Unit` always compiles to `null`, which means that we should've just pass `null` instead

Example:

```fsharp
let myObj = new object()
GC.KeepAlive()
```

Looks quite innocent, but in this case `GC.KeepAlive` does nothing, since it have only single overload `KeepAlive: object? -> void`. Compiler invokes this method as `GC.KeepAlive(null)` and by doing so hides important bug

**Comment by Xyncgas** [comment]
I guess you can pass anything to a method accepting object
A C# method in F# has to be called with ```()``` around the parameters, F# native methods don't have to
